diff --git a/src/main/java/ch/epfl/sweng/javapk/util/Networking.java b/src/main/java/ch/epfl/sweng/javapk/util/Networking.java
index 9f67e71..788e822 100644
--- a/src/main/java/ch/epfl/sweng/javapk/util/Networking.java
+++ b/src/main/java/ch/epfl/sweng/javapk/util/Networking.java
@@ -39,6 +39,15 @@ public class Networking {
                 throw new IOException("HTTP error code: " + responseCode);
             }
 
+            if (target.exists()) {
+                long lastModified = connection.getLastModified();
+
+                if (lastModified > 0 && lastModified < target.lastModified()) {
+                    progress.accept(connection.getContentLengthLong(), connection.getContentLengthLong());
+                    return;
+                }
+            }
+
             // Retrieve the response body as an InputStream.
             stream = connection.getInputStream();
             if (stream != null) {
